cdap-app-fabric/src/main/java/co/cask/cdap/config/DefaultConfigStore.java

```diff
@@ -0,0 +1,220 @@
+          return false;
+        }
+        configTable.table.delete(rowKey(namespace, type, id));
+        return true;
+      }
+    });
+
+    if (!success) {
+      throw new ConfigNotFoundException(namespace, type, id);
+    }
+  }
+
+  @Override
+  public List<Config> list(final String namespace, final String type) {
+    return txnl.executeUnchecked(new TransactionExecutor.Function<ConfigTable, List<Config>>() {
+      @Override
+      public List<Config> apply(ConfigTable configTable) throws Exception {
+        List<Config> configList = Lists.newArrayList();
+        byte[] prefixBytes = rowKeyPrefix(namespace, type);
+        Scanner rows = configTable.table.scan(prefixBytes, Bytes.stopKeyForPrefix(prefixBytes));
+        Row row;
+        while ((row = rows.next()) != null) {
+          Map<String, String> properties = GSON.fromJson(Bytes.toString(row.get(Bytes.toBytes(PROPERTY_COLUMN))),
+                                                         MAP_STRING_STRING_TYPE);
+          configList.add(new Config(getId(row.getRow(), prefixBytes), properties));
+        }
+        return configList;
+      }
+    });
+  }
+
+  @Override
+  public Config get(final String namespace, final String type, final String id) throws ConfigNotFoundException {
+    Config config = txnl.executeUnchecked(new TransactionExecutor.Function<ConfigTable, Config>() {
+      @Override
+      public Config apply(ConfigTable configTable) throws Exception {
+        Row row = configTable.table.get(rowKey(namespace, type, id));
+        if (row.isEmpty()) {
+          return null;
+        }
+        Map<String, String> propertyMap = GSON.fromJson(Bytes.toString(row.get(Bytes.toBytes(PROPERTY_COLUMN))),
+                                                        MAP_STRING_STRING_TYPE);
+        return new Config(id, propertyMap);
+      }
+    });
+
+    if (config == null) {
+      throw new ConfigNotFoundException(namespace, type, id);
+    } else {
+      return config;
+    }
+  }
+
+  @Override
+  public void update(final String namespace, final String type, final Config config) throws ConfigNotFoundException {
+    Boolean success = txnl.executeUnchecked(new TransactionExecutor.Function<ConfigTable, Boolean>() {
+      @Override
+      public Boolean apply(ConfigTable configTable) throws Exception {
+        if (configTable.table.get(rowKey(namespace, type, config.getId())).isEmpty()) {
+          return false;
+        }
+        configTable.table.put(rowKey(namespace, type, config.getId()), Bytes.toBytes(PROPERTY_COLUMN),
+                              Bytes.toBytes(GSON.toJson(config.getProperties())));
+        return true;
+      }
+    });
+
+    if (!success) {
+      throw new ConfigNotFoundException(namespace, type, config.getId());
+    }
+  }
+
+  private String rowKeyString(String namespace, String type, String id) {
+    return String.format("%s%04d%s", rowKeyPrefixString(namespace, type), id.length(), id);
```

Comment:
operate with byte arrays, not strings please. Otherwise hard to read and may be wrong